### PR TITLE
Add ean13 to link_rewrite if available for product combinations in Front Office

### PR DESCRIPTION
**Pull request by matthieumota**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | If we define ean13 for product, it is added to url so change combination with select for example, query to product with attribute redirect to default attribute because compare rewrite and link_rewrite from product. But we need to merge link_rewrite + ean13 in case of.
| Type?         | bug fix
| Category?     | FO
| BC breaks?    | no
| Deprecations? | no
| How to test?  | Just add a product with EAN13 and combination and try to change combination, not work.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/9015)

### CODE REVIEW

File: controllers/front/ProductController.php [hunk 1]
```diff
@@ -53,9 +53,14 @@ class ProductControllerCore extends ProductPresentingFrontControllerCore
     public function canonicalRedirection($canonical_url = '')
     {
         if (Validate::isLoadedObject($this->product)) {
+            $url = $this->product->link_rewrite;
+            if ($this->product->ean13) {
```
Comment (mickaelandrieu) [inline controllers/front/ProductController.php]: I'd prefer:

```diff
- if ($this->product->ean13) {
+ if (!empty($this->product->ean13)
```
The result is the same, but for the people who read this code the first time, they understand that "ean" is not a boolean but probably a string.

**Review by mickaelandrieu — CHANGES_REQUESTED** [review]

File: controllers/front/ProductController.php (hunk 1 above)
Comment (matthieumota) [reply controllers/front/ProductController.php]: I agree with that. I've edited file.

**Comment by mickaelandrieu** [comment]
Thanks @MatthieuMota

**Review by mickaelandrieu — APPROVED** [review]

**Comment by marionf** [comment]
Hello @MatthieuMota 
In which version do you have this issue ?
It has been solved in 1.7.3.1 by this PR: https://github.com/PrestaShop/PrestaShop/pull/7775
And I can't reproduce this issue in develop branch

**Comment by matthieumota** [comment]
Hey @marionf 
Indeed, I confirm that it work's when I update Prestashop :)

**Comment by marionf** [comment]
Ok, thanks for your feedback @MatthieuMota 
I am closing this PR

**Pull request closed without merging**